examples/guides/mp-restful-webservice/README.adoc

```diff
@@ -669,223 +499,210 @@ accepting it. But omitting that validation lets us create an easy health check
-        if (greetResourceError == null) {  // <4>
-            response = Response.ok().build();
-        } else {
-            JsonObject returnObject = JSON.createObjectBuilder()
-                    .add("error", greetResourceError)
-                    .build();
-            response = Response
-                    .status(Response.Status.INTERNAL_SERVER_ERROR)
-                    .entity(returnObject).build();
-        }
-        return response;
-    }
-----
-<1> Implements the path `/alive`.
-<2> It's an HTTP `GET`.
-<3> Gets a `String` describing why the resource should _not_ be considered alive;
-null if the resource is OK.
-<4> Build the `Response` object for return accordingly.
---
-. Add the method to implement `/ready`
+
+. Add the `call` method.
 +
 --
-[source,java]
-// _include::0-5:{health-resource-class}[tag=readyMethod]
-----
-    @SuppressWarnings("checkstyle:designforextension")
-    @Path("/ready")
-    @GET
-    public Response ready() {
-        return Response.ok().build();
-    }
-----
---
-
-=== Update `GreetApplication`
-In the `getClasses` method add `HealthResource.class` to the arguments
-passed to `CollectionsHelper.setOf`:
+The health framework invokes `call` to retrieve the health information associated
+with this health resource.
 
 [source,java]
-// _include::0-6:{greet-application-class}[tags=getClassesMethod]
-----
-    @Override
-    public Set<Class<?>> getClasses() {
-        return CollectionsHelper.setOf(
-                GreetResource.class
-                , HealthResource.class
-        );
+// _include::0-10:{health-resource-class}[tag=callMethod,indent=0]
+----
+public HealthCheckResponse call() {
+    HealthCheckResponseBuilder builder = HealthCheckResponse.builder()
+            .name("greetingAlive"); //<1>
+    if (greeting == null || greeting.getMessage().trim().length() == 0) { //<2>
+        builder.down() //<3>
+               .withData("greeting", "not set or is empty");
+    } else {
+        builder.up(); //<4>
     }
+    return builder.build(); //<5>
+}
 ----
+<1> Create the health check with the name `greetingAlive`.
+<2> The service is alive as long as the greeting message (trimmed) is not empty.
+<3> If not, report that this liveness check is `down` and add an explanatory
+message.
+<4> If so, report that this liveness check is `up`.
+<5> Either way, build the response and return it.
+--
 
-// _include::0-47:{se-guide-adoc}[tags=rebuildAndRerunService;tryReadiness;tryLiveness]
+// _include::0-67:{se-guide-adoc}[tags=rebuildAndRerunService;tryReadiness;!se-HealthChecks-notes]
 === Stop, rebuild and rerun your service
 
 . Stop any running instance of your app.
 . Rebuild the app and then run it.
 
 
-=== Try the readiness check
-Access the readiness check endpoint:
+=== Check the server's health
+Run this command
 [source,bash]
-curl -i -X GET http://localhost:8080/ready
+curl -i -X GET http://localhost:8080/health
+
+and you should see output like this
```

Comment:
You should see output as shown in this example:

Reply:
done